e2e/trust/sign_test.go

```diff
@@ -0,0 +1,80 @@
+package trust
+
+import (
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+const notaryURL = "https://notary-server:4443"
+const registryPrefix = "registry:5000"
+
+const alpineImage = "registry:5000/alpine:3.6"
+const busyboxImage = "registry:5000/busybox:1.27.2"
+const localImage = "registry:5000/signlocal:v1"
+const signImage = "registry:5000/sign:v1"
+
+func TestSignWithTrustSign(t *testing.T) {
+	setupConfigFile(t)
+	defer removeConfigFile(t)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, signImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "-D", "trust", "sign", signImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
```

Comment:
Can we make any assertions about the stdout/stderr here? 

`golden.Assert()` makes it easy to store the expected output in a file and update it when it changes. 

If you expect nothing on stderr (or stdout) you can use `icmd.Expected{Out: icmd.None, Err: icmd.None}` to assert that it's empty. The default is to allow any string.